### ISSUE TITLE
Please create x13s track build

### ISSUE THREAD

**Issue by xnox** [ubuntu/gnome-sdk#113]
For Ubuntu Concept X13s images a customized mesa is needed with adreno fixes to enable accelerated GL on arm64 laptops on Ubuntu desktop.

With the patched gnome-42-2204 snap I get 60FPS in webgl aquarium website inside firefox snap.

I am proposing the following:

1) create x13s track on gnome-42-2204
2) create a new branch x13s-gnome-42-2204  and pull it from https://git.launchpad.net/~ubuntu-concept/gnome-sdk
3) create snap recipe out of that branch (example https://launchpad.net/~ubuntu-concept/+snap/x13s-gnome-42-2204) and publish it to the store x13s track as gnome-42-2204
4) rebuild said snap, whenever latest/edge gnome-42-2204 is rebuilt (as it is staged-snap) and promote to stable on the same cadence as the gnome-42-2204 snap itself

The full proposed snapcraft.yaml can be seen at https://git.launchpad.net/~ubuntu-concept/gnome-sdk/tree/snapcraft.yaml?h=x13s-gnome-42-2204

however the relevant parts are very small:

```yaml
parts:
  gnome:
    plugin: nil
    stage-snaps: [ gnome-42-2204/latest/edge ]
    stage:
      - -usr/lib/$CRAFT_ARCH_TRIPLET/dri
      - -usr/lib/aarch64-linux-gnu/libEGL_mesa.so.0.0.0
      - -usr/lib/aarch64-linux-gnu/libgbm.so.1.0.0
      - -usr/lib/aarch64-linux-gnu/libglapi.so.0.0.0
      - -usr/lib/aarch64-linux-gnu/libwayland-client.so.0
      - -usr/lib/aarch64-linux-gnu/libwayland-server.so.0

  debs:
    after: [ gnome ]
    plugin: nil
    stage-packages:
      - libgbm1
      - libegl-mesa0
      - libglu1-mesa
      - libgl1-mesa-dri
      - mesa-va-drivers
    stage:
      - usr/lib/$CRAFT_ARCH_TRIPLET
```

Effectively use everything from the gnome-42-2204 snap, replace mesa build with the one from the PPA.

Separately, Ubuntu Concept team is working on trying to SRU the required mesa patches into mesa in lunar. I don't know yet if we can SRU them into mesa in jammy.

Alternatively if you don't want to maintain the above build, maybe you can create x13s track and only authorize uploads of this snap into x13s/stable channel maybe? In that case, the snapcraft.yaml above will be tweaked to stage gnome-42-2204 from latest/stable too. If you prefer a more detached approach.

**Comment by kenvandine** [comment]
This has been merged into the gnome-42-2204-adreno branch, LP build created and track has been requested at https://forum.snapcraft.io/t/track-request-for-gnome-42-2204-adreno-track/34714

**Comment by xnox** [comment]
I have enabled this for x13s installs now, and I get 60fps in WebGL Aquarium demo with 1,000 fish now. This is now a success.

Also, Mesa got upgraded in lunar that contains the necessary patches, such that whenever Mesa is upgraded to backported from 23.04 or later, in one of the future gnome-* snaps, the need for such a track will fold.

Thank you very much!
